### ISSUE TITLE
ViewChild in a subclass kills ViewChild in an abstract superclass

### ISSUE THREAD

**Issue by grantbowering** [angular/angular#10461]
**I'm submitting a ...**  (check one with "x")
```
[ ] bug report
[x] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**

If a subclass component has any field marked `@ViewChild`, any other `@ViewChild` field in an abstract superclass will not evaluate.

**Expected/desired behavior**

Both the superclass `@ViewChild` and the subclass `@ViewChild` should be populated.

**Reproduction of the problem**

https://plnkr.co/edit/w5EvslQbHMMVqjx8v52U

The superclass `@ViewChild` _*does populate*_ from the subclass template, but only if the subclass `@ViewChild` is commented out.  

If uncommented, the subclass `@ViewChild` is populated, but the superclass one is not.

**What is the expected behavior?**

Both the superclass `@ViewChild` and the subclass `@ViewChild` should be populated.

**What is the motivation / use case for changing the behavior?**

Multiple components that inherit common code but supply their own template.  The linked plunkr demonstrates the use case where the abstract class encapsulates all the code and logic for the common feature of performing a search, while the child component provides the specific template in which the `#searchInput` element can be found and whatever other divergent behavior.

**Please tell us about your environment:**

* **Angular version:** 2.0.0-rc.4
 
* **Browser:** Chrome 51
 
* **Language:** TypeScript 1.8

**Comment by alexzuza** [comment]
I think it related with https://github.com/angular/angular/issues/7968#issuecomment-219865739

**Comment by grantbowering** [comment]
This is exactly what I was looking for, thank you! :smile: 

I guess part of the source of my surprise is that the `@ViewChild` decorator from the base class **_does_** get inherited if there's no `@ViewChild` in the subclass... If decorator inheritance wasn't happening, I wouldn't have expected it to work at **_all_**; there's no `@Component` decorator on the base class, and/thus no template or anything, so the fact that the base `@ViewChild` even works at all by itself and does get populated with an element reference from the super template is counter-intuitive.  It does make sense why it gets superseded by the subclass's now that I read the discussion in #7968 though. 

(I also got caught off guard by it because the first implementer component I wrote didn't have its own `@ViewChild`, and then in the second implementer component, it wasn't the implementer's that was breaking but the abstract's, which was already working in the first implementer component, so it took me a moment to even realize the implementer's `@ViewChild` was what was breaking it!)

The discussion in #7968 is mostly about inheriting the `@Component` decorator or other class-level `annotations` decorators.  I wouldn't push for that and could see it being a big headache.  I'm also ambivalent about `parameters` and `design:paramtypes` decorators.  But I would say that `propMetadata` decorators (`@ViewChild` and `@ViewChildren`, `@ContentChild` and `@ContentChildren`, `@Input` and `@Output`, `@HostBinding` and `@HostListener`) both have more use cases for inheritance, and are possibly at least from a strategy perspective relatively straight-forward to merge (just smush them all together, and if anything has the same name, lowest subclass wins)...  I'm sure I'm not even considering a whole bunch of complicated edge cases though.  At the least, I'm glad to have this thread around for people to google up if the simple-behavior-for-simple-cases @alexzuza provided above is what they need!  

So I'm not sure if I should close the ticket, but I'll definitely change it to a feature request at least.
